packages/expect/src/asymmetricMatchers.ts

```diff
@@ -178,11 +178,21 @@ class ObjectContaining extends AsymmetricMatcher<Record<string, unknown>> {
       return true;
     } else {
       for (const property in this.sample) {
-        const expected =
-          typeof this.sample[property] === 'object' &&
-          !(this.sample[property] instanceof AsymmetricMatcher)
-            ? objectContaining(this.sample[property] as Record<string, unknown>)
-            : this.sample[property];
+        let expected = this.sample[property];
+
+        if (typeof this.sample[property] === 'object') {
+          const samplePropertyPrototype = Object.getPrototypeOf(
+            this.sample[property],
+          );
+          if (
+            samplePropertyPrototype === Object.prototype ||
```

Comment:
I _think_ this might break for cases hit by #2549, but 🤞 it does not

Reply:
If it does break, afaik the next best ways to check for "plain objects" are `Object.getPrototypeOf(sample)?.constructor?.name === 'Object'` and `Object.prototype.toString.call(sample) ...`. These are imprecise, but at least they work across realms.

`Array.isArray()` could be substituted for the `Array.prototype` check to fix that case.

Reply:
I think I can confirm that this is broken for cross-realm inputs, like in #2549. https://github.com/ninevra/jest/commit/0553005ae367b20e2fe5c670e65c67c0e4fa5f6b adds a failing test on top of this PR to demonstrate the issue.